src/convertToObject.js

```diff
@@ -3,16 +3,33 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const validatedStyles = sourceString.split(';')
+    .map(style => style.split(':'))
+    .filter(element => element.length !== 1)
+    .reduce(callback, {});
+
+  return validatedStyles;
+}
+
+function callback(acc, prop) {
+  const styleName = prop[0].trim();
+  const styleValue = prop[1].trim();
+
+  return {
+    ...acc,
+    [styleName]: styleValue,
+  };
 }
```

Comment:
```suggestion
function callback(acc, [key, value]) {
  const styleName = key[0].trim();
  const styleValue = value[1].trim();

  return {
    ...acc,
    [styleName]: styleValue,
  };
}
```

Reply:
You can use array destructuring here

Reply:
then we don't need to use a value in brackets(after key and value), right?

  const styleName = key.trim();
  const styleValue = value.trim();